p2p/host/resource-manager/conn_limiter.go

```diff
@@ -110,12 +218,52 @@ func (cl *connLimiter) addConn(ip netip.Addr) bool {
 func (cl *connLimiter) rmConn(ip netip.Addr) {
 	cl.mu.Lock()
 	defer cl.mu.Unlock()
-	limits := cl.connLimitPerCIDRIP4
-	countsPerLimit := cl.ip4connsPerLimit
+	cidrLimits := cl.cidrLimitV4
+	connsPerCidr := cl.connsPerCIDRV4
+	limits := cl.connLimitPerSubnetV4
+	connsPerLimit := cl.ip4connsPerLimit
 	isIP6 := ip.Is6()
 	if isIP6 {
-		limits = cl.connLimitPerCIDRIP6
-		countsPerLimit = cl.ip6connsPerLimit
+		cidrLimits = cl.cidrLimitV6
+		connsPerCidr = cl.connsPerCIDRV6
+		limits = cl.connLimitPerSubnetV6
+		connsPerLimit = cl.ip6connsPerLimit
+	}
+
+	// Check CIDR limits first
+
+	if len(connsPerCidr) == 0 && len(cidrLimits) > 0 {
+		// Initialize just in case. We should have already initialized in
+		// addConn, but if the callers calls rmConn first we don't want to panic
+		connsPerCidr = make([]int, len(cidrLimits))
+	}
+	for i, limit := range cidrLimits {
+		if limit.Network.Contains(ip) {
+			count := connsPerCidr[i]
+			if count <= 0 {
+				log.Errorf("unexpected conn count for ip %s. Was this not added with addConn first?", ip)
```

Comment:
Should there be a `return` here?

Reply:
Yeah, I think that makes sense as an effort to reduce the unexpectedness of the unexpected case.